packages/ui/src/components/RadioGroup/index.tsx

```diff
@@ -0,0 +1,150 @@
+import styled from '@emotion/styled'
+import {
+  type ComponentProps,
+  type InputHTMLAttributes,
+  type ReactNode,
+  createContext,
+  useContext,
+  useMemo,
+} from 'react'
+import { Radio } from '../Radio'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+
+type RadioGroupContextType = {
+  groupName: string
+  groupValue: string | number
+} & Required<Pick<InputHTMLAttributes<HTMLInputElement>, 'onChange'>> &
+  Pick<InputHTMLAttributes<HTMLInputElement>, 'required'>
+
+const RadioGroupContext = createContext<RadioGroupContextType | undefined>(
+  undefined,
+)
+
+type RadioGroupRadioProps = Omit<
+  ComponentProps<typeof Radio>,
+  'onChange' | 'checked' | 'required'
+>
+
+const RadioGroupRadio = ({
+  onFocus,
+  onBlur,
+  disabled,
+  error,
+  name,
+  value,
+  label,
+  helper,
+  className,
+  autoFocus,
+  onKeyDown,
+  'data-testid': dataTestId,
+}: RadioGroupRadioProps) => {
+  const context = useContext(RadioGroupContext)
+
+  if (!context) {
+    throw new Error('RadioGroup.Radio can only be used inside a RadioGroup')
+  }
+
+  const { groupName, onChange, groupValue } = context
+
+  const radioName = `${groupName}-${name ?? ''}`
```

Comment:
If no name is provided, why not generating one?

Reply:
By the way can't we make it nested, like `${groupName}.${name ?? ''}` so in the form the value of the radio will be inside of the radio group name